tests/e2e/specs/admin-tracking.test.js

```diff
@@ -39,77 +38,56 @@ describe( 'Providing client configuration', () => {
-		const analyticsScriptTag = await page.$x(
-			'//script[contains(@src,"https://www.google-analytics.com/analytics.js")]'
-		);
-		expect( analyticsScriptTag.length ).not.toEqual( 0 );
+		await expect( page ).toMatchElement( 'script[src^="https://www.google-analytics.com/analytics.js"]' );
 
 		// Ensure tag manager script tag exists.
-		const tagManagerScriptTag = await page.$x(
-			'//script[contains(@src,"https://www.googletagmanager.com/gtag/js?id=UA-130569087-3")]'
-		);
-		expect( tagManagerScriptTag.length ).not.toEqual( 0 );
-
+		await expect( page ).toMatchElement( 'script[src^="https://www.googletagmanager.com/gtag/js?id=UA-130569087-3"]' );
 	} );
 
 	it( 'should uncheck opt-in box when clicked', async() => {
 
 		await page.waitForSelector( '#opt-in' );
 
-		await expect( page ).toClick( '#opt-in' );
-
-		await page.waitForResponse( res => {
-			const reqURL = new URL( res.url() );
-
-			return '/wp-json/wp/v2/settings' === reqURL.pathname;
-		} );
+		await Promise.all( [
+			page.waitForResponse( res => res.url().match( 'wp/v2/settings' ) ),
+			expect( page ).toClick( '#opt-in' ),
+		] );
 
 		await page.waitForSelector( '.mdc-checkbox:not(.mdc-checkbox--selected) #opt-in' );
 
 		// Ensure unchecked checkbox exists.
-		const optinUnChecked = await page.$( '.mdc-checkbox:not(.mdc-checkbox--selected) #opt-in' );
-		expect( optinUnChecked.length ).not.toEqual( 0 );
+		await expect( page ).toMatchElement( '.mdc-checkbox:not(.mdc-checkbox--selected) #opt-in' );
 	} );
 
 	it( 'should not have tracking code when not opted in', async() => {
 		await page.waitForSelector( '#opt-in' );
 
-		await expect( page ).toClick( '#opt-in' );
-
-		await page.waitForResponse( res => {
-			const reqURL = new URL( res.url() );
-
-			return '/wp-json/wp/v2/settings' === reqURL.pathname;
-		} );
+		await Promise.all( [
+			page.waitForResponse( res => res.url().match( 'wp/v2/settings' ) ),
+			expect( page ).toClick( '#opt-in' ),
+		] );
 
 		await page.reload();
 
 		// Ensure unchecked checkbox exists.
-		const optinUnChecked = await page.$( '.mdc-checkbox:not(.mdc-checkbox--selected) #opt-in' );
-		expect( optinUnChecked.length ).not.toEqual( 0 );
+		await expect( page ).toMatchElement( '.mdc-checkbox:not(.mdc-checkbox--selected) #opt-in' );
 
 		// Ensure no analytics script tag exists.
-		const analyticsScriptTag = await page.$x(
-			'//script[contains(@src,"https://www.google-analytics.com/analytics.js")]'
-		);
-		expect( analyticsScriptTag.length ).toEqual( 0 );
+		await expect( page ).not.toMatchElement( 'script[src^="https://www.google-analytics.com/analytics.js"]' );
 
 		// Ensure no tag manager script exists.
-		const tagManagerScriptTag = await page.$x(
-			'//script[contains(@src,"https://www.googletagmanager.com/gtag/js?id=UA-130569087-3")]'
-		);
-		expect( tagManagerScriptTag.length ).toEqual( 0 );
+		await expect( page ).not.toMatchElement( 'script[src^="https://www.googletagmanager.com/gtag/js?id=UA-130569087-3"]' );
```

Comment:
Why did you make all these changes to the tracking test? How do they relate to the Search Console property being required for permission?

Reply:
These were quick changes to bring this test up to date with the patterns and utilities we're using which are mostly for consistency and readability. It isn't directly related to this requirement but we have been updating some of the older E2E tests as we touch them. If you'd prefer this to go through another issue we can back out any non-critical changes.